source/gui/_addonStoreGui/viewModels/store.py

```diff
@@ -313,22 +347,122 @@ def _handleEnableDisable(self, listItemVM: AddonListItemVM[_AddonManifestModel],
 		listItemVM.status = getStatus(listItemVM.model)
 		self.refresh()
 
-	def enableOverrideIncompatibilityForAddon(self, listItemVM: AddonListItemVM[_AddonManifestModel]) -> None:
+	def enableOverrideIncompatibilityForAddon(
+			self,
+			listItemVM: AddonListItemVM[_AddonManifestModel],
+			askConfirmation: bool = True,
+			useRememberChoiceCheckbox: bool = False,
+	) -> tuple[bool, bool]:
 		from ... import mainFrame
-		if _shouldEnableWhenAddonTooOldDialog(mainFrame, listItemVM.model):
+		if askConfirmation:
+			res, shouldRememberChoice = _shouldEnableWhenAddonTooOldDialog(
+				mainFrame,
+				listItemVM.model,
+				
+				useRememberChoiceCheckbox=useRememberChoiceCheckbox,
+			)
+		else:
+			res = True
+			shouldRememberChoice = True
+		if res:
 			listItemVM.model.enableCompatibilityOverride()
 			self._handleEnableDisable(listItemVM, True)
+		return res, shouldRememberChoice
 
 	def enableAddon(self, listItemVM: AddonListItemVM) -> None:
 		self._handleEnableDisable(listItemVM, True)
 
+	def enableAddons(self, listItemVMs: Iterable[AddonListItemVM[_AddonStoreModel]]) -> None:
+		shouldEnableIncompatible = True
+		shouldRememberChoice = False
+		for aVM in listItemVMs:
+			if aVM.status in (
+				AvailableAddonStatus.INCOMPATIBLE_DISABLED,
+				AvailableAddonStatus.PENDING_INCOMPATIBLE_DISABLED,
+			) and not aVM.model.isCompatible and aVM.model.canOverrideCompatibility:
+				if shouldRememberChoice:
+					if shouldEnableIncompatible:
+						self.enableOverrideIncompatibilityForAddon(aVM, askConfirmation=False)
+					else:
+						log.debug(
+							f"Skipping {aVM.Id} as override incompatibility has been previously declined for all remaining"
+							" add-ons."
+						)
+				else:
+					shouldEnableIncompatible, shouldRememberChoice = self.enableOverrideIncompatibilityForAddon(
+						aVM,
+						askConfirmation=True,
+						useRememberChoiceCheckbox=True,
+					)
+			elif aVM.status in (
+				AvailableAddonStatus.DISABLED,
+				AvailableAddonStatus.PENDING_DISABLE,
+			):
+				self.enableAddon(aVM)
+			else:
+				log.debug(
+					f"Skipping {aVM.Id} as it is not disabled, pending disable, incompatible disabled "
+					"or pending incompatible disabled"
+				)
+
+	
+
 	def disableAddon(self, listItemVM: AddonListItemVM) -> None:
 		self._handleEnableDisable(listItemVM, False)
 
-	def replaceAddon(self, listItemVM: AddonListItemVM) -> None:
+	def disableAddons(self, listItemVMs: Iterable[AddonListItemVM[_AddonStoreModel]]) -> None:
+		for aVM in listItemVMs:
+			if aVM.status not in [

```

Comment:
this doesn't quite match `canUseDisableAction`. Should we just use that here?

Reply:
Good catch!